// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package optimization_guide.proto;

import "components/optimization_guide/proto/features/common_quality_data.proto";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.features.proto";

option java_outer_classname = "FormsAnnotationsProto";

// DO NOT EDIT THIS FILE DIRECTLY!
//
// This file is generated in g3 and then synced to Chrome. Instead, please
// refer to http://go/chrome-intelligence-feature-protos (Google-internal link),
// and then changes will be synced with Chrome automatically.

message FormsAnnotationsLoggingData {
  FormsAnnotationsRequest request = 1;

  FormsAnnotationsResponse response = 2;
}

message FormsAnnotationsRequest {
  // The context of the page that the form is on.
  PageContext page_context = 1;

  // The form to build annotations for.
  FormData form_data = 4;

  // The current entries in the user annotations database.
  repeated UserAnnotationsEntry entries = 5;
}

message FormsAnnotationsResponse {
// The user annotations entries that were newly added to the database.
  repeated UserAnnotationsEntry added_entries = 1;

  // The user annotations entry IDs that were not changed.
  repeated int64 unchanged_entry_ids = 2;

  // The user annotations entry IDs that were deleted.
  repeated int64 deleted_entry_ids = 3;
}
